src/OrchardCore/OrchardCore.Search.Elasticsearch.Core/Services/ElasticIndexingService.cs

```diff
@@ -53,193 +61,173 @@ public class ElasticIndexingService
             _indexingTaskManager = indexingTaskManager;
             _elasticConnectionOptions = elasticConnectionOptions.Value;
             _siteService = siteService;
+            _contentManager = contentManager;
+            _contentItemIndexHandlers = contentItemIndexHandlers;
+            _store = store;
             _contentDefinitionManager = contentDefinitionManager;
             _logger = logger;
         }
 
-        public async Task ProcessContentItemsAsync(string indexName = default)
+        public async Task ProcessContentItemsAsync(params string[] indexNames)
         {
             if (!_elasticConnectionOptions.FileConfigurationExists())
             {
                 return;
             }
 
+            var indexSettingsList = (await _elasticIndexSettingsService.GetSettingsAsync()).ToList();
+
+            if (indexNames != null && indexNames.Length > 0)
+            {
+                indexSettingsList = indexSettingsList
+                    .Where(x => indexNames.Contains(x.IndexName, StringComparer.OrdinalIgnoreCase))
+                    .ToList();
+            }
+
+            if (indexSettingsList.Count == 0)
+            {
+                return;
+            }
+
             var allIndices = new Dictionary<string, long>();
             var lastTaskId = long.MaxValue;
-            IEnumerable<ElasticIndexSettings> indexSettingsList = null;
 
-            if (string.IsNullOrEmpty(indexName))
```

Comment:
Here, when we passed an empty indexName it meant that we wanted to update all indices. This is something you removed.

Reply:
That function is still there. If you pass no parameters, we'll update all indexes